commons/src/main/java/io/hyscale/commons/utils/ResourceSelectorUtil.java

```diff
@@ -47,4 +48,12 @@ public static String getSelector(String appName) {
         return getSelectorFromLabelMap(ResourceLabelBuilder.build(appName));
     }
 
+    /**
+	 *it will return namespace field selector
+	 */
+    public static String getNamespaceSelector(String namespace) {
+    	StringBuilder fieldSelector = new StringBuilder();
+		fieldSelector.append(MetadataFieldSelector.METADATA.getName()).append(ToolConstants.DOT).append(MetadataFieldSelector.NAMESPACE.getName()).append(ToolConstants.EQUALS).append(namespace);
```

Comment:
The MetaDataFieldSelector should have metadata.namespace !!
If you see our Resourcelabelkey is not being build everytime , the label is just defined similarly the field should be defined.